uxarray/grid.py

```diff
@@ -371,3 +372,111 @@ def integrate(self, var_ds, quadrature_rule="triangular", order=4):
         integral = np.dot(face_areas, face_vals)
 
         return integral
+
+    def _populate_cartesian_xyz_coord(self):
+        """A helper function that populates the xyz attribute in UXarray.ds.
+        This function is called when we need to use the cartesian coordinates
+        for each node to do the calculation but the input data only has the
+        "Mesh2_node_x" and "Mesh2_node_y" in degree.
+
+        Note
+        ----
+        In the UXarray, we abide the UGRID convention and make sure the following attributes will always have its
+        corresponding units as stated below:
+
+        Mesh2_node_x
+         unit:  "degree_east" for longitude
+        Mesh2_node_y
+         unit:  "degrees_north" for latitude
+        Mesh2_node_z
+         unit:  "m"
+        Mesh2_node_cart_x
+         unit:  "m"
+        Mesh2_node_cart_y
+         unit:  "m"
+        Mesh2_node_cart_z
+         unit:  "m"
+        """
+
+        # Check if the cartesian coordinates are already populated
+        if "Mesh2_node_cart_x" in self.ds.keys():
+            return
+
+        # check for units and create Mesh2_node_cart_x/y/z set to self.ds
+        num_nodes = self.ds.Mesh2_node_x.size
```

Comment:
Number of nodes is already stored as an attribute in the Grid class: `self.nMesh2_node`